backend/app/controllers/api/v3/users_controller.rb

```diff
@@ -32,6 +32,342 @@ module V3
+      #       name: <name>,
+      #       login: <login>,
+      #       permission_ids: <permission_ids>,
+      #       user_email: <email>,
+      #       user_phone: <phone>
+      #     }
+      #   }
+      #
+      # @!method show_info(token, id)
+      # @param token [String] a token
+      # @param id [Int] the id of the user
+      def show_info
+        # Check for admin permissions
+        status, response = check_token_for_permission(1)
+        render json: response.to_json, status: status.to_sym and return if response[:error]
+
+        # Get user
+        id = Input.int(params[:id])
+        user = User.find_id_show_info(id)
+
+        render json: {
+          user: user
+        }.to_json, status: :ok
+      end
+
+      # Create a new user.
+      #
+      # <b>API Call:</b>
+      #   POST: /api/v3/users/create
+      #   {
+      #     token: <token>
+      #     user: {
+      #       name: <name>,
+      #       login: <login>,
+      #       password: <password>,
+      #       permission_ids: [ <array_of_permission_ids> ]
+      #     }
+      #   }
+      #
+      # <b>API Return Success:</b>
+      #   STATUS CODE: 201
+      #   {
+      #     user: {
+      #       id: <user_id>,
+      #       name: <name>,
+      #       login: <login>,
+      #       permission_ids: <permission_ids>
+      #     }
+      #   }
+      #
+      # @!method create(token, user)
+      # @param token [String] a token
+      # @param user [Hash] the user
+      def create
+        # Check for admin permissions
+        status, response = check_token_for_permission(1)
+        render json: response.to_json, status: status.to_sym and return if response[:error]
+
+        # Read sample type parameter
+        params_user = params[:user] || {}
+
+        # Create sample type
+        user, errors = User.create(params_user)
+        render json: { errors: errors }.to_json, status: :ok and return if !user
+
+        render json: { user: user }.to_json, status: :created
+      end
+
+      # Update a user's info.
+      #
+      # <b>API Call:</b>
+      #   POST: /api/v3/users/<id>/update_info
+      #   {
+      #     token: <token>
+      #     id: <user_id>,
+      #     user: {
+      #       name: <name>
+      #       email: <email>
+      #       phone: <phone>
+      #     }
+      #   }
+      #
+      # <b>API Return Success:</b>
+      #   STATUS CODE: 200
+      #   {
+      #     user: {
+      #       id: <user_id>,
+      #       name: <name>,
+      #       login: <login>,
+      #       permission_ids: <permission_ids>
+      #     }
+      #   }
+      #
+      # @!method update_info(token, id, user)
+      # @param token [String] a token
+      # @param id [Int] the id of the user
+      # @param user [Hash] the user
+      def update_info
+        # Check for admin permissions <or> self
+        params_user_id = Input.int(params[:id])
+        status, response = check_token_for_permission(-1, params_user_id)
```

Comment:
I agree in principle. Let's discuss. 

Some notes: At the end of the day I need to pass the user_id of the person you are looking up to know whether it is yourself.  It only really happens for this API call (i.e., editing your own information), so I can go back to the single lookup and then make a second call when necessary.  It is a bit sub-optimal performance-wise, but since it only happens here (i.e., editing your own information), then it is not that big a hit to take.

Let's look at it again after I revert back to the old code and add a second call to check for oneself.

Reply:
Refactored to use 2 calls.  Much cleaner even if there are now 2 separate calls, and only affects a few very seldomly used places in the code (specifically only when an admin is changing someone else's profile info) so not a big deal.